extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java

```diff
@@ -313,6 +314,15 @@ private OpenAPI generateAnnotationModel(IndexView indexView, Capabilities capabi
 
         String defaultPath = config.getValue("quarkus.http.root-path", String.class);
 
+        Optional<String> resteasyPath = config.getOptionalValue("quarkus.resteasy.path", String.class);
+        if (resteasyPath.isPresent()) {
+            if (defaultPath == null || defaultPath.equals("/")) {
+                defaultPath = resteasyPath.get();
+            } else {
+                defaultPath += resteasyPath.get();
+            }
+        }
+
```

Comment:
So I'm a bit concerned about that one.

What I'm worried about is what if RESTEasy is mixed with something else? AFAICS, you add the RESTEasy path as a prefix for everything? I might not be totally aware of how it works underneath but I find it suspicious.

Also, I think we should tackle the SPI issue now and move `ResteasyJaxrsConfigBuildItem` to `extensions/resteasy-server-common/spi`. Maybe we could deprecate the old one for now and create a new one. We could generate both for a while.

And we would have the OpenAPI extension depend on `quarkus-resteasy-server-common-spi`.

Reply:
It is mixed up because I thought it should work like the swagger-ui.path. For example, if you set the ```quarkus.http.root-path=/test``` and the ```quarkus.swagger-ui.path=/swagger```, the Swager UI is available at /test/swagger. AFAICS it is the same for smallrye-health for example, so I did it that way for concistency.

Reply:
That wasn't my issue. What I was wondering is if it's not dependent on RESTEasy, I suppose it supports other things. And thus what if we have RESTEasy and that other thing? Should the path include the RESTEasy part?

That's more a question for @phillip-kruger . Your path is OK, even if I would prefer we do the SPI thing.

Reply:
Hi @gsmet - The other thing is Spring-Web, that I do not think has a property like this a.f.a.i.k. Not sure if it needs to apply for Spring endpoints if you have both Spring and JAX-RS. But also at the moment SmallRye can not handle that.